gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapperTest.java

```diff
@@ -0,0 +1,197 @@
+        var ehrExtract = unmarshallEhrExtract("immunization_with_missing_optional_values.xml");
+        List immunizationList = immunizationMapper.mapToImmunization(ehrExtract, getPatient(), getEncounterList());
+
+        var immunization = (Immunization) immunizationList.get(0);
+        assertMissingData(immunization, immunizationList);
+    }
+
+    @Test
+    public void mapObservationToImmunizationWithMultipleObservationStatements() {
+        var ehrExtract = unmarshallEhrExtract("full_valid_immunization_with_multiple_observation_statements.xml");
+        List immunizationList = immunizationMapper.mapToImmunization(ehrExtract, getPatient(), getEncounterList());
+
+        var immunization = (Immunization) immunizationList.get(0);
+        assertThat(immunizationList.size()).isEqualTo(THREE);
+        assertThat(immunization.getId()).isEqualTo(OBSERVATION_ROOT_ID);
+        assertThat(immunization.getMeta().getProfile().get(0).getValue()).isEqualTo(META_PROFILE);
+        assertThatIdentifierIsValid(immunization.getIdentifierFirstRep(), immunization.getId());
+    }
+
+    @Test
+    public void mapObservationToImmunizationWithEffectiveTimeCenter() {
+        var ehrExtract = unmarshallEhrExtract("immunization_with_only_center_effective_time.xml");
+        List immunizationList = immunizationMapper.mapToImmunization(ehrExtract, getPatient(), getEncounterList());
+
+        var immunization = (Immunization) immunizationList.get(0);
+        assertImmunizationWithHighEffectiveTimeCenter(immunization);
+    }
+
+    @Test
+    public void mapObservationToImmunizationWithEffectiveTimeLow() {
+        var ehrExtract = unmarshallEhrExtract("immunization_with_only_low_effective_time.xml");
+        List immunizationList = immunizationMapper.mapToImmunization(ehrExtract, getPatient(), getEncounterList());
+
+        var immunization = (Immunization) immunizationList.get(0);
+        assertImmunizationWithEffectiveTimeLow(immunization);
+    }
+
+    @Test
+    public void mapObservationToImmunizationWithHighAndLowEffectiveTime() {
+        var ehrExtract = unmarshallEhrExtract("immunization_with_high_and_low_effective_time.xml");
+        List immunizationList = immunizationMapper.mapToImmunization(ehrExtract, getPatient(), getEncounterList());
+
+        var immunization = (Immunization) immunizationList.get(0);
+        assertImmunizationWithHighAndLowEffectiveTime(immunization);
+    }
+
+    @Test
+    public void mapObservationToImmunizationWithHighEffectiveTime() {
+        var ehrExtract = unmarshallEhrExtract("immunization_with_only_high_effective_time.xml");
+        List immunizationList = immunizationMapper.mapToImmunization(ehrExtract, getPatient(), getEncounterList());
+
+        var immunization = (Immunization) immunizationList.get(0);
+        assertImmunizationWithHighEffectiveTime(immunization);
+    }
+
+    private void assertImmunizationWithHighAndLowEffectiveTime(Immunization immunization) {
```

Comment:
Some tests are timezone dependant - they're failing if timezone is different than GMT.